compiler/rustc_data_structures/Cargo.toml

```diff
@@ -22,7 +22,7 @@ stable_deref_trait = "1.0.0"
 rayon = { version = "0.3.0", package = "rustc-rayon" }
 rayon-core = { version = "0.3.0", package = "rustc-rayon-core" }
 rustc-hash = "1.1.0"
-smallvec = { version = "1.0", features = ["union", "may_dangle"] }
+smallvec = { version = "1.6.1", features = ["union", "may_dangle"] }
```

Comment:
These changes are not necessary. Version 1.6.1 is semver compatible with 1.0, so `cargo update -p smallvec` is enough to update it.